### PR TITLE
Docs for output formatters

### PR DESCRIPTION
**Pull request by GiudGiud**
Improves documentation to Executor and Output actions as well as input file formatters. 

Refs #18333 #17566 #16248 #15968

### CODE REVIEW

File: framework/doc/content/source/actions/AddOutputAction.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 ## Overview
 
-An [MooseObjectAction](MooseObjectAction.md) that acts on the existence of the `[Outputs]` block
+This [MooseObjectAction](MooseObjectAction.md) act on the `[Outputs]` block
```
Comment (GiudGiud) [inline framework/doc/content/source/actions/AddOutputAction.md:7]:
```suggestion
This [MooseObjectAction](MooseObjectAction.md) acts on the `[Outputs]` block
```

File: framework/doc/content/source/actions/ReadExecutorParamsAction.md [hunk 2]
```diff
@@ -1,21 +1,8 @@
...
-## Overview
-
-!! Replace these lines with information regarding the ReadExecutorParamsAction action.
-
-## Example Input File Syntax
-
-!! Describe and include an example of how to use the ReadExecutorParamsAction action.
-
-!syntax description /Executors/ReadExecutorParamsAction
+This [MooseObjectAction](MooseObjectAction.md) act on the `[Executor]` block
```
Comment (GiudGiud) [inline framework/doc/content/source/actions/ReadExecutorParamsAction.md:5]:
```suggestion
This [MooseObjectAction](MooseObjectAction.md) acts on the `[Executor]` block
```

File: framework/doc/content/source/actions/ReadExecutorParamsAction.md [hunk 3]
```diff
@@ -1,21 +1,8 @@
...
+to create Executor objects. See [syntax/Executors/index.md] for more information.
```
Comment (cticenhour) [inline framework/doc/content/source/actions/ReadExecutorParamsAction.md:6]: We have the `AddExecutorAction` to create Executor objects though, right? This just reads in the executor options and adds automatic preconditioning if required, it looks like. If this is the case, then the source class description might need some updating too.

File: framework/doc/content/source/outputs/formatters/YAMLFormatter.md [hunk 4]
```diff
@@ -1,6 +1,72 @@
...
+
+This formatter is used for the `--yaml` command line option, described
+[here](modules/doc/content/application_usage/command_line_usage.md optional=True). It is also available for the
+application developer, either directly or using an [AdvancedOutput](syntax/Outputs/index.md#advanced-output).
+
+## Example output
+
+We reproduce here the output of the `InputParameters` YAML dump for the [Functions](syntax/Functions/index.md)
+block and the [MooseParsedFunction.md]. We can see metadata about each parameter, such as their
+C++ type, their default or description.
```
Comment (cticenhour) [inline framework/doc/content/source/outputs/formatters/YAMLFormatter.md:14]:
```suggestion
C++ type, their default, or their description.
```

File: framework/doc/content/source/outputs/formatters/InputFileFormatter.md [hunk 5]
```diff
@@ -1,6 +1,137 @@
...
+file syntax.
+
+This formatter is used for the `--show-input` command line option, described
+[here](modules/doc/content/application_usage/command_line_usage.md optional=True). It is also available for the
+application developer, either directly or using an [AdvancedOutput](syntax/Outputs/index.md#advanced-output).
+
+## Example output
+
+The snippet below shows the output for `--show-input` for the boundary condition and executioner
+blocks of the simple_diffusion test, shown further below. We can see additional metadata about each object,
```
Comment (cticenhour) [inline framework/doc/content/source/outputs/formatters/InputFileFormatter.md:13]:
```suggestion
blocks of the `simple_diffusion` test, shown further below. We can see additional metadata about each object,
```

**Review by cticenhour — CHANGES_REQUESTED** [review]
Only a couple comments

File: framework/doc/content/source/actions/ReadExecutorParamsAction.md (hunk 3 above)
Comment (cticenhour) [reply framework/doc/content/source/actions/ReadExecutorParamsAction.md]: Really weird how you couldn't get the markdown syntax to work with `AddExectorAction`. I suppose it's enough of a base class that MooseDocs doesn't fully register it. Regardless I like your new changes.

**Review by cticenhour — APPROVED** [review]

File: framework/doc/content/source/actions/ReadExecutorParamsAction.md (hunk 3 above)
Comment (GiudGiud) [reply framework/doc/content/source/actions/ReadExecutorParamsAction.md]: yeah. I tried to get the automatic stub generation to get it right but it didnt work on my machine. I ll try again somewhere else another time. Thanks for the review!

**Pull request merged**